src/transcription/liveTranscription.ts

```diff
@@ -36,6 +36,9 @@ export class LiveTranscription extends EventEmitter {
     };
 
     this._socket.onclose = (event: WebSocket.CloseEvent) => {
+      const newTarget: any = event.target;
+      const dgErrorIndex = newTarget["_req"].res.rawHeaders.indexOf("dg-error");
+      event.reason = newTarget["_req"].res.rawHeaders[dgErrorIndex + 1];
```

Comment:
I felt like these had a chance to fail(with the help of TS), so I did some testing. 

I was able to artificially trigger some errors by using an invalid URL. `newTarget["_req"]` is `null` on a 404, resulting in the following: `TypeError: Cannot read property 'res' of null`. Not sure if there are other scenarios in which the `_req` field is `null`, but maybe worth some checks in here to prevent error's in that sphere

Maybe there's a reason TS is angry about the signature `newTarget` as a `WebSocket` :smile:

Reply:
Interesting. Maybe I'll set up a conditional check and assign it if it is available?